doc/source/security/using-kmip.rst

```diff
@@ -0,0 +1,78 @@
+.. _using-kmip:
+
+==========================================================
+Using the Key Management Interoperability Protocol (KMIP)
+==========================================================
+
+This feature is **technical preview** quality.
+
+**Percona Server for MySQL** 8.0.27-18 adds support for the `OASIS Key Management Interoperability Protocol (KMIP) <https://docs.oasis-open.org/kmip/kmip-spec/v2.0/os/kmip-spec-v2.0-os.html>`__. This implementation was tested with the `PyKMIP server <https://pykmip.readthedocs.io/en/latest/server.html>`__ and the `HasiCorp Vault Enterprise KMIP Secrets Engine <https://www.vaultproject.io/docs/secrets/kmip>`__.
+
+KMIP enables communication between key management systems and the database server. The protocol can do the following:
+
+* Streamline encryption key management
+
+* Eliminate redundant key management processes
+
+Component installation
+--------------------------------------
+
+The KMIP component must be installed with a manifest. A keyring component is not loaded with the ``--early-plugin-load`` option on the server. The server uses a manifest and the component consults its configuration file during initialization. You should only load a keyring component with a manifest file. Do not use the ``INSTALL_COMPONENT`` statement, which loads the keyring components too late in the startup sequence of the server. For example, ``InnoDB`` requires the component, but because the components are registered in the ``mysql.component`` table, this table is loaded after ``InnoDB`` initialization. 
+
+You should create a global manifest file named ``mysqld.my`` in the installation directory and, optionally, create a local manifest file, also named ``mysqld.my`` in a data directory.
+
+To install a keyring component, you must do the following:
+
+1. Write a manifest in a valid JSON format
+2. Write a configuration file
+
+A manifest file indicates which component to load. During startup, the server can read either a global manifest file, or a global manifest file that points to a local manifest file.
+
+The server reads the global manifest file from the installation directory. If the global manifest file points to a local manifest file, the server also reads the local manifest file in the data directory. If the manifest file does not exist, the server does not load the component associated with that file.
+
+A local manifest file allow loading different components for multiple instances of the server. The loading instructions are specific to that data directory. Different server instances can use different keyring components.
```

Comment:
A local manifest file ALLOWS loading different components for multiple instances ON the server.

Reply:
Rewritten.